includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -999,19 +1002,54 @@ private function get_disallowed_attributes_in_node( DOMElement $node, $attr_spec
 		return $attrs_to_remove;
 	}
 
+	/**
+	 * Adapt width and height to conform to the AMP layout.
+	 *
+	 * @param DOMElement $node Node.
+	 */
+	private function adapt_dimensions_to_layout( DOMElement $node ) {
+		$layout = $node->getAttribute( 'layout' );
+
+		if ( ! amp_is_attribute_empty( $layout ) ) {
+			$width  = $node->getAttribute( 'width' );
+			$height = $node->getAttribute( 'height' );
+
+			// If the width & height are `100%` the layout must be `fill`.
+			if ( '100%' === $width && '100%' === $height ) {
```

Comment:
This condition and the condition below need to run even when `layout` is not set, as they imply the required `layout`.